### DISCUSSION TITLE
Is this pandoc or a pandoc wrapper?

### DISCUSSION THREAD

**Discussion by buhtz** [boisgera/pandoc#54] [Q&A] [answered]
It is not clear for me from reading the code or the docu.

Is this repo the official pandoc or is it a pandoc wrapper? In the latter case I need a pandoc executable/binary installed on my system right?

**Comment by boisgera** [comment, marked as answer]
Hi @buhtz ,

This is not the official pandoc repo, which is https://github.com/jgm/pandoc. I was expecting my [README](https://github.com/boisgera/pandoc#pandoc---python-library) to be good enough to disambiguate this issue, but maybe I was too optimistic 😕: 

> 🚀 Getting started
> --------------------------------------------------------------------------------
>
> [Pandoc] – the general markup converter (and Haskell library) written by [John MacFarlane] – needs to be available. 
> You may follow the official [installation instructions][pandoc-install] or use [conda]:
>
>    $ conda install -c conda-forge pandoc
>
> [...]
>
>  [Pandoc]: https://pandoc.org/
> [John MacFarlane]: https://johnmacfarlane.net/
> [pandoc-install]: https://pandoc.org/installing.html
> [conda]: https://docs.conda.io
>

Anyway, if you search for "pandoc Python" on Github, here is what you get:

![Screenshot from 2023-03-22 14-28-44](https://user-images.githubusercontent.com/1217694/226919622-9495c4c0-7071-4c1b-9067-fb5c327c7ea3.png)

  - [jgm/pandocfilters](https://github.com/jgm/pandocfilters) requires the pandoc CLI  and it's official (@jgm being the author of pandoc). But it's not a wrapper of the Pandoc CLI ; it's a Python module to support the development of Pandoc filters in Python. E.g. it helps you to write a `caps.py` Python script to capitalize all words in a document. You use it like this:

    ```bash
    pandoc -t json -s | ./caps.py | pandoc -f json
    ```
  - [Panflute](http://scorreia.com/software/panflute/) is an unofficial but (arguably) more Pythonic library to design Pandoc filters in Python. Since it provides filters, the pandoc CLI is still necessary to do anything useful with it.

  - [Pandoc (Python library)](https://github.com/boisgera/pandoc#pandoc---python-library)  (this project) is similar to Panflute in many ways ; it also tries to be Pythonic, it also requires the Pandoc CLI to be installed, it also focuses on document transformations, etc. However, it doesn't use (or provide) filters ; your document transform will be a Python program.

  - [JessicaTegner/pypandoc](https://github.com/JessicaTegner/pypandoc) is a wrapper of the pandoc CLI. It requires the pandoc CLI, but its API can even help you with this (see https://github.com/JessicaTegner/pypandoc#installing-pandoc).

Does it help?

Cheers,

Sébastien

**Reply by buhtz** [reply]
Thanks a lot for explaining.

> > [Pandoc](https://pandoc.org/) – the general markup converter (and Haskell library) written by [John MacFarlane](https://johnmacfarlane.net/) – needs to be available.

Maybe it depends on my none-native English that this sentences isn't 100% clear to me.

Something like this (from PyPandas) is more clear to someone like me:

> Pypandoc uses pandoc, so it needs an available installation of pandoc.

**Reply by boisgera** [reply]
You're right, my sentence is convoluted ... In my defence, English is not my native langage either! 😊
